### PR TITLE
feat: correctly sort bazel module versions

### PR DESCRIPTION
**Pull request by kormide**
Sort module versions in the official order.

See https://github.com/bazelbuild/bazel-central-registry/pull/1431#discussion_r1477936416.

### CODE REVIEW

File: src/domain/version.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    if (typeof a.value !== typeof b.value) {
+      if (typeof a.value === "number") {
+        return -1;
+      } else {
+        return 1;
+      }
+    }
+
+    if (typeof a.value === "string") {
+      return a.value.localeCompare(b.value as string);
```
Comment (fmeum) [inline src/domain/version.ts:76]: Should we make this locale independent?

**Review by fmeum — APPROVED** [review]

**Comment by cgrindel** [comment]
The code looks good. However, I wanted to point out that [a TypeScript library for bzlmod verions](https://github.com/hobofan/bzlmod-version/blob/main/src/bzlmod-version.ts#L201). Do you think it makes sense to leverage the library?

**Comment by kormide** [comment]
> a TypeScript library for bzlmod verions

Nice, I had no idea this existing. I'll use that instead.

**Comment by kormide** [comment]
> > a TypeScript library for bzlmod verions
> 
> Nice, I had no idea this existed. I'll use that instead.

It looks like this hasn't actually been published to npm, yet. @hobofan Are there any plans to do so?

**Pull request merged**